sqlglot/lineage.py

```diff
@@ -276,8 +276,44 @@ def to_node(
                 reference_node_name=reference_node_name,
                 trim_selects=trim_selects,
             )
+
+        elif scope.pivots:
+            pivot = next((p for p in scope.pivots if p.alias_or_name == c.table), None)
```

Comment:
This needs to check `p.unpivot` too, because `scope.pivots` contains `Pivot` expressions that also represent the `UNPIVOT` operator.